common/app/model/Cors.scala

```diff
@@ -28,4 +27,14 @@ object Cors extends Results with implicits.Requests {
       case None => result
     }
   }
+
+  private[model] def isWhitelisted(origin: String, corsOrigins: Seq[String], extraWhitelist: Seq[String]): Boolean = {
```

Comment:
This will of course work, but this function is a bit dirty in the way that it breaks the ideal abstraction gradiant (abstraction should increase with depth).

It basically takes some abstract 'String' parameters and make them less abstract (more concrete) by turning them into URIs (what if the 'origin' is not in a URI format?), and by assuming these sequences of strings would be list of hosts, effectively making this function much more specialised than its signature (regardless of names) would suggest.

A way to fix that would be to align the parameter types with what it actually expects (so being of type (URI, Seq[Host], Seq[Host]) -> Boolean ) .

Reply:
Notice that `origin` is compared to `corsOrigins` differently than it is compared to `extraWhitelist`. Passing `origin` as URI would mean having to change it back to a String to compare it against the items in `corsOrigins`. Also `corsOrigin` is supplied by `conf.Configuration.ajax` as a String as is used as a String throughout the codebase so don't see the goal in changing types there. As for the distinction between `corsOrigin` String format vs `extraWhitelist` String format that's very well documented in the tests.